File: program36.c
#include<stdio.h>
#include<stdbool.h>


////////////////////////////////////////////////////////////////////////////////
//
//	Name : ChkPrime
//	Input : Integer
//	Output : Boolean
//	Description : Checks Whether Input Integer is Prime or not
//	Author : Kaushal Sachin Kapadnis
//  Date : 	26/4/2022
//
////////////////////////////////////////////////////////////////////////////////


bool ChkPrime(int iNo)
{
	int iCnt = 0;
		
	for(iCnt = 2; iCnt <=  (iNo / 2);iCnt++)
	{
		if(iNo % iCnt == 0)
		{
			break;
		}
	}
	
	if(iCnt == ((iNo / 2) + 1))
	{
		return true;
	}
	else
	{
		return false;
	}
}

//////////////////////////////////////////////////////////////////////////////////
//	Entry Pint Function
//////////////////////////////////////////////////////////////////////////////////

int main()
{
	int iValue = 0;
	bool bRet = false;
	
	printf("Enter Number ");
	scanf("%d",&iValue);
	
	bRet = ChkPrime(iValue);
	
	if(bRet == true)
	{
		printf("%d is prime number ",iValue);
	}
	else
	{
		printf("%d is not prime number ",iValue);
	}
	
	return 0;
}

File: program36.java
import java.lang.*;
import java.io.*;

class Numbers
{
	public int iNo;
	public boolean bFlag;
	
	public Numbers(int x)
	{
		this.iNo = x;
		this.bFlag = true;
	}
	
	public boolean ChkPrime()
	{
		int iCnt = 0;
		
		for(iCnt = 2; iCnt < (this.iNo); iCnt++)
		{
			if((this.iNo % iCnt ) == 0)
			{
				this.bFlag = false;
				break;
			}
		}
		
	return this.bFlag;
	}
}

class program36
{
	public static void main(String arg[])
	{
		InputStreamReader iobj = new InputStreamReader(System.in);
		BufferedReader bobj = new BufferedReader(iobj);
		
		int iValue = 0;
		boolean bRet = false; 
		
		System.out.println("Enter Number : ");
		try
		{
			iValue = Integer.parseInt(bobj.readLine());
		}
		catch(IOException obj)
		{
			
		}
		
		Numbers nobj = new Numbers(iValue);
		
		bRet = nobj.ChkPrime();
		
		if(bRet == true)
		{
			System.out.println(iValue + " is a prime number");
		}
		else
		{
			System.out.println(iValue + " is not a prime number");
		}
		
		
	}
}

// Time Complexity : O(N)
// where N is no of Digit